besu/src/main/java/org/hyperledger/besu/services/RpcEndpointServiceImpl.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.services;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.JsonRpcMethod;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.PluginJsonRpcMethod;
+import org.hyperledger.besu.plugin.services.RpcEndpointService;
+import org.hyperledger.besu.plugin.services.rpc.PluginRpcRequest;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class RpcEndpointServiceImpl implements RpcEndpointService {
+  private final Map<String, Function<PluginRpcRequest, ?>> rpcMethods = new HashMap<>();
+
+  @Override
+  public <T> void registerRPCEndpoint(
+      final String namespace,
+      final String functionName,
+      final Function<PluginRpcRequest, T> function) {
+    checkArgument(namespace.matches("\\p{Alnum}+"), "Namespace must be only alpha numeric");
+    checkArgument(functionName.matches("\\p{Alnum}+"), "Function Name must be only alpha numeric");
+    checkNotNull(function);
+
+    rpcMethods.put(namespace + "_" + functionName, function);
+  }
```

Comment:
what prevents us from overriding a built-in rpc endpoint?  Do we consider that a feature?

Reply:
Nothing is preventing us from overriding existing endpoints. Couldn't think of a good reason to stop someone from doing it.

Reply:
resolved with https://github.com/hyperledger/besu/pull/2754/commits/fc65ddfc44fa95cc42d8f8629524cddd0de767f0 preventing users to override methods. We can easily delete the code in the future if desired.